File: client_server/models/JobDetail.js
import mongoose from "mongoose";

const JobDetailSchema = new mongoose.Schema({
  url: {
    type: String,
    required: true,
    unique: true,
    index: true,
  },
  thumbnail: {
    type: String,
  },
  job_title: {
    type: String,
    required: true,
  },
  company_url: {
    type: String,
  },
  company_name: {
    type: String,
  },
  province: {
    type: String,
    required: true,
  },
  salary: {
    type: String,
  },
  skills: {
    type: [String],
    default: [],
  },
  descriptions: {
    type: Map,
    of: String,
  },
  job_info: {
    type: Map,
    of: String,
  },
  collected_at: {
    type: Date,
    default: Date.now,
  },
});

const JobDetail = mongoose.models.JobDetail || mongoose.model("JobDetail", JobDetailSchema, "JobDetail");

export default JobDetail;

File: scraper/src/model/job_detail.py
from datetime import datetime, timezone


def now():
    return datetime.now(timezone.utc).isoformat()


class JobDetail:
    def __init__(
        self,
        url: str,
        thumbnail=None,
        job_title=None,
        company_url=None,
        company_name=None,
        province=None,
        salary=None,
        skills=None,
        descriptions=None,
        job_info=None,
    ):
        self.url = url
        self.thumbnail = thumbnail
        self.job_title = job_title
        self.company_url = company_url
        self.company_name = company_name

        self.province = province
        self.salary = salary
        self.skills = skills
        self.descriptions = descriptions
        self.job_info = job_info

        self.collected_at = now()
